code/game/objects/items/weapons/RCD.dm

```diff
@@ -221,7 +221,7 @@ RCD
 					if(do_after(user,A, 50))
 						if(!useResource(decon_cost, user)) return 0
 						activate()
-						A:ChangeTurf(A:under_turf)
+						A:ChangeTurf(get_base_turf(A.z))

```

Comment:
Can't fix that, that would be adminbuse (if the base_turf was made to be floor or wall, respectively)